### PR TITLE
Feature/replace deprecated body

### PR DESCRIPTION
**Pull request by mstrYoda**
In this pr the old `body()` method calls replaced with `requestBody().asBytes()` method which both exists in `Request` class. The intention is to remove deprecated code and keep source code clean.

### CODE REVIEW

File: core/src/main/java/feign/Request.java [hunk 1]
```diff
@@ -190,17 +190,6 @@ public Charset charset() {
...
   }
 
-  /**
-   * If present, this is the replayable body to send to the server. In some cases, this may be
-   * interpretable as text.
-   *
-   * @see #charset()
-   * @deprecated use {@link #requestBody()} instead
-   */
-  public byte[] body() {
```
Comment (velo) [inline core/src/main/java/feign/Request.java]: I don't recall when we deprecated this.

If was on 9.x think is ok to drop it.

Comment (mstrYoda) [reply core/src/main/java/feign/Request.java]: If deprecated method is completely removed from usage, is not that ok to remove the method itself? I think removing it prevents further usage. Should I keep it or remove it?

Comment (velo) [reply core/src/main/java/feign/Request.java]: A `decoder` may exists out there that still uses it.

Comment (mstrYoda) [reply core/src/main/java/feign/Request.java]: I see.. Sorry I am not fully familiar on source code. I'll put it back.

Just a question out of this context, is it acceptable to use a 3rd party library (like apache commons lang) for some requirement?

Comment (velo) [reply core/src/main/java/feign/Request.java]: core has being clean from external dependencies and we are most likely to want to keep it that way.

Particularly about apache commons-*, every time I see them I read "old deprecated code that somehow still survived" =)

**Comment by kdavisk6** [comment]
@mstrYoda 

And one more small housekeeping note, can you please update the description of this PR with the following format:

```
A short description of the change, your intention, and what you are trying to solve.

Fixes [issue number]
```

This helps us keep track of what issue the PR is for and provides us with background on your thinking and the intended outcome of this change.

**Comment by mstrYoda** [comment]
> @mstrYoda
> 
> And one more small housekeeping note, can you please update the description of this PR with the following format:
> 
> ```
> A short description of the change, your intention, and what you are trying to solve.
> 
> Fixes [issue number]
> ```
> 
> This helps us keep track of what issue the PR is for and provides us with background on your thinking and the intended outcome of this change.

I updated description.

**Comment by kdavisk6** [comment]
@mstrYoda 

Thanks for your contribution!

**Pull request merged**